addons/arsenal/functions/fnc_verifyLoadout.sqf

```diff
@@ -23,220 +23,289 @@ if (count _loadout == 2) then {
+                                            INC(_unavailableItemsAmount);
                                         };
+                                    } else {
+                                        _nullItemsList pushBackUnique _item;
+                                        ((_loadout select _dataIndex) select 1) set [_forEachIndex, []];
+                                        INC(_nullItemsAmount);
                                     };
-
-                                    case 3: {
-                                        private _item = _x select 0;
-
-                                        if (isClass (_magCfg >> _item)) then {
-                                            if !(
-                                                    _item in (GVAR(virtualItems) select IDX_VIRT_ITEMS_ALL) ||
-                                                    _item in (GVAR(virtualItems) select 15) ||
-                                                    _item in (GVAR(virtualItems) select 16)
-                                                ) then {
-
-                                                _unavailableItemsList pushBackUnique _item;
-                                                ((_loadout select _dataIndex) select 1) set [_currentIndex, []];
-                                                _unavailableItemsAmount = _unavailableItemsAmount + 1;
-                                            };
-                                        } else {
-
-                                            _nullItemsList pushBackUnique _item;
-                                            ((_loadout select _dataIndex) select 1) set [_currentIndex, []];
-                                            _nullItemsAmount = _nullItemsAmount + 1;
+                                };
+                                // Weapons have 2 entries: Weapon info array and amount
+                                case (_x isEqualTypeArray [[], 0]): {
+                                    [_x select 0] call _fnc_weaponCheck;
+                                };
+                                // Misc. items have 2 entries: Name and amount, containers have 2 entries: Name and isBackpack
+                                default {
+                                    _x params ["_item"];
+
+                                    // Check if item exists
+                                    if (
+                                        isClass (_cfgWeapons >> _item) ||
+                                        {isClass (_cfgMagazines >> _item)} ||
+                                        {isClass (_cfgGlasses >> _item)} ||
+                                        {isClass (_cfgVehicles >> _item)}
+                                    ) then {
+                                        // Check if item is available in arsenal
+                                        if !(
+                                            ////////////////// use _item in GVAR(virtualItemsFlat) ?
```

Comment:
Could `GVAR(virtualItemsFlat)` be used instead of all of the individual comparisons? The individual comparisons don't look in the weapons and magazines - which it would for `GVAR(virtualItemsFlat)` - would that be a problem?

Reply:
Should be fine.

Reply:
Updated to work with `GVAR(virtualItemsFlat)`.